src/components/calcite-rating/calcite-rating.tsx

```diff
@@ -193,7 +193,16 @@ export class CalciteRating {
   //--------------------------------------------------------------------------
   @Method()
   async setFocus(): Promise<void> {
-    this.el.querySelector("input").focus();
+    const {
+      el: { shadowRoot, value }
+    } = this;
+    if (value > 0) {
+      const selected = shadowRoot.querySelectorAll(".selected");
+      const lastSelectedId = selected[selected.length - 1].getAttribute("for");
+      focusElement(shadowRoot.querySelector(`input[id="${lastSelectedId}"]`));
+    } else {
+      focusElement(shadowRoot.querySelector("fieldset input"));
```

Comment:
nitpick: you could use refs instead of querySelector on line 202 and 204 to grab a hold of the nodes in a more direct way.  I think using refs is more performant too, based on some previous review comments I had when I used `querySelector` in the past.

Reply:
Good idea, I'll try that - thanks @eriklharper!